### PR TITLE
fix: generate mgmake import rather then having it in magefile

### PR DESCRIPTION
**Pull request by viktorvoltaire**
The whole `.mage` is built on `mgmake`. We build a binary, then we run the `mgmake` target. So its mandatory, but really only during the generate step. 

Therefor I think it doesnt make sense to have it in the magefiles, since it might not be clear for the end user why it should be there in the first place, removing it would break the entire setup without really explaining why. And it makes the actual magefiles just include the expected targets

### CODE REVIEW

**Review by Edholm — APPROVED** [review]
LGTM!

File: .mage/tools.mk [hunk 1]
```diff
@@ -2,16 +2,30 @@ mage_folder := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
 mage_generated_path := $(mage_folder)/gen
 mage_targets_file := $(mage_generated_path)/targets.mk
 mage := $(mage_generated_path)/local-mage
+mgmake := $(mage_folder)/mgmake.go
+
+define mgmake_content
+// +build mage
```
Comment (odsod) [inline .mage/tools.mk:8]: do we want a Go 1.17 build directive here too?

File: example/.mage/tools.mk [hunk 2]
```diff
@@ -2,16 +2,30 @@ mage_folder := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
...
+define mgmake_content
+// +build mage
+
+package main
+
+// mage:import
+import _ "go.einride.tech/mage-tools/mgmake"
+endef
+
+export mgmake_content
```
Comment (odsod) [inline example/.mage/tools.mk:16]: why is this needed?

Comment (viktorvoltaire) [reply example/.mage/tools.mk:16]: From what i understood while reading the Makefile documentaiton, it was needed to make `mgmake_content` a usable variable

Comment (odsod) [reply example/.mage/tools.mk:16]: ah! is that specific to `define`? (never used that before)

Comment (viktorvoltaire) [reply example/.mage/tools.mk]: Yeah, first time for me too. I gave it a try without the export, and it seems to be working fine without it :)

Comment (odsod) [reply example/.mage/tools.mk]: woppa!

File: .mage/tools.mk (hunk 1 above)
Comment (viktorvoltaire) [reply .mage/tools.mk:9]: Yep, fixed!

**Review by odsod — APPROVED** [review]
Shiny! :star:

**Comment by viktorvoltaire** [comment]
> Shiny! star

One thing, that I now realize while doing a quick test. Is that any dependency that this target brings in, gets added to `go.mod`, meaning if for some reason a user would do a manual `go mod tidy` they would get a diff. It doesnt have an affect when we generate stuff but it could cause some confusion. Rather then deleting it, should we just generate it if it doesnt exist. And add some comments to the file, similar to `wire_gen.go`?

**Comment by viktorvoltaire** [comment]
I pushed my new suggestion :innocent:

File: example/.mage/tools.mk [hunk 3]
```diff
@@ -2,16 +2,30 @@ mage_folder := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
...
 $(mage_targets_file): $(mage_folder)/go.mod $(mage_folder)/*.go
 	@git clean -fdx $(mage_generated_path)
 	@mkdir -p $(mage_generated_path)
+	$(file > $(mgmake),$(mgmake_content))
 	@cd $(mage_folder) && \
 		go mod tidy && \
 		go run main.go -compile $(mage) && \
-		$(mage) generateMakefile $(@)
+		$(mage) generateMakefile $(@) && \
+		rm -f $(mgmake)
```
Comment (odsod) [inline example/.mage/tools.mk:28]: does this need to be re-generated?

Comment (odsod) [reply example/.mage/tools.mk:28]: it shouldn't be here anymore, right?

Comment (viktorvoltaire) [reply example/.mage/tools.mk:28]: Right :see_no_evil:

**Review by odsod — APPROVED** [review]
